src/main/java/org/dcsa/bkg/controller/EventController.java

```diff
@@ -0,0 +1,98 @@
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.server.reactive.ServerHttpRequest;
+import org.springframework.http.server.reactive.ServerHttpResponse;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import javax.validation.ConstraintViolationException;
+import javax.validation.Valid;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.Size;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@RestController
+@Validated
+@RequestMapping(
+    value = "events",
+    produces = {MediaType.APPLICATION_JSON_VALUE})
+public class EventController extends AbstractEventController<BKGEventService, Event> {
+
+  private final BKGEventService bkgEventService;
+
+  public EventController(@Qualifier("BKGEventServiceImpl") BKGEventService bkgEventService) {
+    this.bkgEventService = bkgEventService;
+  }
+
+  @Override
+  public BKGEventService getService() {
+    return this.bkgEventService;
+  }
+
+  @Override
+  public String getType() {
+    return "Event";
+  }
+
+  @Override
+  protected ExtendedRequest<Event> newExtendedRequest() {
+    return new ExtendedGenericEventRequest(extendedParameters, r2dbcDialect) {
+      @Override
+      public void parseParameter(Map<String, List<String>> params) {
+        Map<String, List<String>> p = new HashMap<>(params);
+        // Add the eventType parameter (if it is missing) in order to limit the resultset
+        // to *only* SHIPMENT events
+        p.putIfAbsent("eventType", List.of(EventType.SHIPMENT.name()));
+        super.parseParameter(p);
+      }
+    };
+  }
+
+  @GetMapping
+  public Flux<Event> findAll(
+      @RequestParam(value = "shipmentEventTypeCode", required = false)
+          @ValidEnum(clazz = ShipmentEventTypeCode.class)
+          String shipmentEventTypeCode,
+      @RequestParam(value = "documentTypeCode", required = false)
+          @ValidEnum(clazz = DocumentTypeCode.class)
+          String documentTypeCode,
+      @RequestParam(value = "carrierBookingReference", required = false) @Size(max = 35)
+          String carrierBookingReference,
+      @RequestParam(value = "transportDocumentReference", required = false) @Size(max = 20)
+          String transportDocumentReference,
+      @RequestParam(value = "transportDocumentTypeCode", required = false)
+          @ValidEnum(clazz = TransportDocumentTypeCode.class)
+          String transportDocumentTypeCode,
+      @RequestParam(value = "limit", defaultValue = "1", required = false) @Min(1) int limit,
+      ServerHttpResponse response,
+      ServerHttpRequest request) {
+    return super.findAll(response, request);
+  }
+
+  @Override
+  public Mono<Event> create(@Valid @RequestBody Event event) {
```

Comment:
Don't we need to add something similar to updates? It is not allowed to update an event... ;-)

Reply:
...or for that matter delete an event

Reply:
Yup makes sense, I will add that as well. This is also needed in OVS and TNT too then.